File: ios/Classes/plugin/domain/repositories/BluetoothRepository.swift
import Foundation

public protocol BluetoothRepository {
    func scanDevices(completion: @escaping ([BluetoothDeviceEntity]) -> Void)
    func connectToDevice(address: String) -> Bool
    func disconnectFromDevice() -> Bool
    func printData(data: String, size: Int, align: Int, bold: Bool) -> Bool
    func printEmptyLine(callTimes: Int) -> Bool
}

File: android/src/main/kotlin/com/maktubcompany/easy_blue_printer/plugin/domain/repository/BluetoothRepository.kt
package com.maktubcompany.easy_blue_printer.plugin.domain.repository

import com.maktubcompany.easy_blue_printer.plugin.domain.entities.BluetoothDeviceEntity

interface BluetoothRepository {
    fun getPairedDevices(): List<BluetoothDeviceEntity>
    fun connectToDevice(address: String): Boolean
    fun disconnectToDevice(): Boolean
    fun printData(data: String, size: Int, align: Int, bold: Boolean): Boolean
    fun printEmptyLine(callTimes: Int): Boolean
    fun isConnected(): Boolean
}